src/panel/poi/ActionButtons.jsx

```diff
@@ -1,62 +1,58 @@
-import classnames from 'classnames';
+
+import Button from 'src/components/ui/Button';
 
 export default class ActionButtons extends React.Component {
   static propTypes = {
     poi: PropTypes.object.isRequired,
     isDirectionActive: PropTypes.bool,
     openDirection: PropTypes.func,
     openShare: PropTypes.func,
-    isPhoneNumberVisible: PropTypes.bool,
-    showPhoneNumber: PropTypes.func,
+    onClickPhoneNumber: PropTypes.func,
     isPoiInFavorite: PropTypes.bool,
     toggleStorePoi: PropTypes.func.isRequired,
   }
 
   renderPhone() {
-    if (!this.props.isPhoneNumberVisible) {
-      return <button className="poi_panel__action icon-icon_phone poi_phone_container_hidden"
-        onClick={this.props.showPhoneNumber}
-      >
-        <div>{_('Show number', 'poi')}</div>
-      </button>;
-    }
-    return <a className="poi_panel__action icon-icon_phone poi_phone_container_revealed"
+    const { onClickPhoneNumber, poi } = this.props;
+    return <Button
+      className="poi_panel__action__button poi_panel__action__phone"
+      onClick={onClickPhoneNumber}
+      icon="icon_phone"
+      href={poi.blocksByType.phone.url}
       rel="noopener noreferrer external"
-      href={this.props.poi.blocksByType.phone.url}
-    >
-      <div>{this.props.poi.blocksByType.phone.local_format}</div>
-    </a>;
+    />;
   }
 
   render() {
     const shouldRenderPhone = this.props.poi.blocksByType && this.props.poi.blocksByType.phone;
 
     return <div className="poi_panel__actions">
-      <button className={classnames(
-        'poi_panel__action',
-        'poi_panel__actions__icon__store',
-        {
-          'icon-icon_star-filled': this.props.isPoiInFavorite,
-          'icon-icon_star': !this.props.isPoiInFavorite,
-        })
-      }
-      onClick={this.props.toggleStorePoi}
-      >
-        <div>{this.props.isPoiInFavorite ? _('Saved', 'poi') : _('Favorites', 'poi')}</div>
-      </button>
-      <button className="poi_panel__action icon-share-2" onClick={this.props.openShare}>
-        <div>{_('Share', 'poi')}</div>
-      </button>
       {this.props.isDirectionActive &&
-        <button className="poi_panel__action icon-corner-up-right"
-          onClick={this.props.openDirection}>
-          <div>{_('Directions', 'poi')}</div>
-        </button>
+        <Button
+          className="button--noBorder poi_panel__action__direction"
+          variant="invert"
+          onClick={this.props.openDirection}
+        >
+          { _('Directions', 'poi panel') }
+        </Button>
       }
+
       {shouldRenderPhone && this.renderPhone()}
+
+      <Button
+        className="poi_panel__action__button poi_panel__action__favorite"
+        onClick={this.props.toggleStorePoi}
+        icon={this.props.isPoiInFavorite ? 'icon_star-filled' : 'star'}
+      />
+
+      <Button
+        className="poi_panel__action__button poi_panel__action__share"
+        onClick={this.props.openShare}
+        icon="share-2"
+      />
```

Comment:
minor issue: a rectangular border is visible on click on these buttons, on Firefox:
![image](https://user-images.githubusercontent.com/4726554/82472801-0028cf80-9ac9-11ea-9718-413fbdbb3259.png)

Reply:
It's possible to remove it with a special rule for Firefox:
```css
button::-moz-focus-inner {
  border: 0;
}
```

But I think there are other a11y/interactivity problems with the buttons here, like no hover / or active state.

This can be addressed separately I guess, probably by copying the button styles from the design system.

Reply:
Ok, I added your proposed fix @bbecquet

Reply:
So I guess we'll address the question of hover/active states separately :)